hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationRegistry.java

```diff
@@ -106,7 +106,8 @@ long getLastCallId() {
     public boolean register(Invocation invocation) {
         final long callId;
         try {
-            callId = callIdSequence.next(invocation.op.isUrgent());
+            boolean force = invocation.op.isUrgent() || invocation.isRetryCandidate();
```

Comment:
Only executor thread(s) in InvocationMonitor is allowed to submit retries. So, once registry is full, it can be exceed only by number of retrying threads. It's 1 right now.